sklearn/metrics/pairwise.py

```diff
@@ -1271,6 +1271,13 @@ def pairwise_distances_chunked(X, Y=None, reduce_func=None,
             X_chunk = X[sl]
         D_chunk = pairwise_distances(X_chunk, Y, metric=metric,
                                      n_jobs=n_jobs, **kwds)
+        if ((X is Y or Y is None)
+                and PAIRWISE_DISTANCE_FUNCTIONS.get(metric, None)
+                is euclidean_distances):
```

Comment:
Should we be introducing this for all metrics, perhaps with a FutureWarning where we detect the diagonal is not close to 0?

Reply:
in both chunked and non-chunked? How do we know the callable we get passed is actually a metric? Should we try and detect and warn if it's not? Maybe the deprecation will be enough of a warning.